pennylane/operation.py

```diff
@@ -873,11 +875,55 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
                 f"{len(self._wires)} wires given, {self.num_wires} expected."
             )
 
+        self._check_batching(params)
+
         self.data = list(params)  #: list[Any]: parameters of the operator
 
         if do_queue:
             self.queue()
 
+    def _check_batching(self, params):
+        """Check if the expected numbers of dimensions of parameters coincides with the
+        ones received and sets the ``_batch_size`` attribute.
+
+        Args:
+            params (tuple): Parameters with which the operator is instantiated
+
+        The check always passes and sets the ``_batch_size`` to ``None`` for the default
+        `Operator.ndim_params` property but subclasses may overwrite it to define fixed
```

Comment:
~If we expect subclasses to overwrite this method, should it be public?~ Nevermind, I think I misread!

Reply:
```suggestion
        ``Operator.ndim_params`` property but subclasses may overwrite it to define fixed
```

Reply:
Thanks! :)